pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -208,6 +214,23 @@ func (i *RegistriesConf) finish() error {
 	return nil
 }
 
+func (i *RegistriesConf) validateReleaseImageIsSameInRegistriesConf(source string) error {
+
+	releaseImage := &releaseimage.Image{}
+	releaseImage.Generate(asset.Parents{})
+
+	releaseImagePath := strings.Split(releaseImage.PullSpec, ":")[0]
+
+	if !strings.Contains(string(i.File.Data), releaseImagePath) {
+		if source == "Generated" {
```

Comment:
nit: I'd prefer a boolean (or named constants) over magic strings.

Reply:
I would even prefer not to have such distinction at all, and making the validation step completely unaware of the source

Reply:
Restructured but I think the source field is needed especially if we want to deliver different error messages in the case of Generated Vs Loaded.